### ISSUE TITLE
Docker Image throwes Java Error

### ISSUE THREAD

**Issue by morzan1001** [dependency-check/DependencyCheck#4385]
Hi, I've been having the problem for a while now that the dependency scanner I use in my gitlab pipeline fails. I'm using the latest Docker image and the container is thrown away after each run of the pipeline, so I don't think the database is compromized or anything. I have also tried removing all volumes and containers on the runner, but the error still persists. Here is the output of the dependency scanner:

```
$ /usr/share/dependency-check/bin/dependency-check.sh --project Name --out ./reports/ --format JSON --format HTML --scan . --enableExperimental --failOnCVSS 7
java.util.zip.ZipException: invalid entry CRC (expected 0xc030c281 but got 0x55798fab)
	at java.base/java.util.zip.ZipInputStream.read(ZipInputStream.java:223)
	at java.base/java.util.zip.ZipInputStream.closeEntry(ZipInputStream.java:142)
	at org.h2.util.Utils.loadResource(Utils.java:420)
	at org.h2.util.Utils.getResource(Utils.java:386)
	at org.h2.message.DbException.<clinit>(DbException.java:74)
	at org.h2.message.TraceObject.<clinit>(TraceObject.java:108)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at org.owasp.dependencycheck.data.nvdcve.DatabaseManager.initialize(DatabaseManager.java:187)
	at org.owasp.dependencycheck.data.nvdcve.DatabaseManager.<init>(DatabaseManager.java:125)
	at org.owasp.dependencycheck.data.nvdcve.CveDB.<init>(CveDB.java:242)
	at org.owasp.dependencycheck.Engine.openDatabase(Engine.java:995)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:868)
	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:683)
	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:609)
	at org.owasp.dependencycheck.App.runScan(App.java:256)
	at org.owasp.dependencycheck.App.run(App.java:188)
	at org.owasp.dependencycheck.App.main(App.java:83)
[INFO] Checking for updates
[INFO] NVD CVE requires several updates; this could take a couple of minutes.
[INFO] Download Started for NVD CVE - 2002
[INFO] Download Complete for NVD CVE - 2002  (919 ms)
[INFO] Processing Started for NVD CVE - 2002
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.fasterxml.jackson.module.afterburner.util.MyClassLoader (file:/usr/share/dependency-check/lib/jackson-module-afterburner-2.13.2.jar) to method java.lang.ClassLoader.findLoadedClass(java.lang.String)
WARNING: Please consider reporting this to the maintainers of com.fasterxml.jackson.module.afterburner.util.MyClassLoader
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Download Started for NVD CVE - 2003
```
[...]
```
[INFO] Processing Complete for NVD CVE - Modified  (1858 ms)
[INFO] Begin database maintenance
[INFO] Updated the CPE ecosystem on 124831 NVD records
[INFO] Removed the CPE ecosystem on 3499 NVD records
[INFO] End database maintenance (20250 ms)
[INFO] Begin database defrag
[INFO] End database defrag (8002 ms)
[INFO] Check for updates complete (147688 ms)
[INFO] 
Dependency-Check is an open source tool performing a best effort analysis of 3rd party dependencies; false positives and false negatives may exist in the analysis performed by the tool. Use of the tool and the reporting provided constitutes acceptance for use in an AS IS condition, and there are NO warranties, implied or otherwise, with regard to the analysis or its use. Any use of the tool and the reporting provided is at the user’s risk. In no event shall the copyright holder or OWASP be held liable for any damages whatsoever arising out of or in connection with the use of this tool, the analysis performed, or the resulting report.
   About ODC: https://jeremylong.github.io/DependencyCheck/general/internals.html
   False Positives: https://jeremylong.github.io/DependencyCheck/general/suppression.html
💖 Sponsor: https://github.com/sponsors/jeremylong
[INFO] Analysis Started
[INFO] Finished File Name Analyzer (0 seconds)
[INFO] Finished MSBuild Project Analyzer (0 seconds)
[INFO] Finished Dependency Merging Analyzer (0 seconds)
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Finished Hint Analyzer (0 seconds)
[ERROR] Exception occurred initializing NPM CPE Analyzer.
[ERROR] Exception occurred initializing CPE Analyzer.
[INFO] Finished False Positive Analyzer (0 seconds)
[INFO] Finished NVD CVE Analyzer (0 seconds)
[INFO] Finished Sonatype OSS Index Analyzer (1 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (4 seconds)
[INFO] Writing report to: /builds/name/name/./reports/dependency-check-report.json
[INFO] Writing report to: /builds/name/name/./reports/dependency-check-report.html
[ERROR] An exception occurred accessing the database
[ERROR] An exception occurred accessing the database
```

Maybe I'm missing something fundamental here, but maybe someone has an idea what I can try to get the scanner running again. Thanks in any case!

**Comment by aikebah** [comment]
To get the scanner running again you need to clear the local nvd caches as the current NVD stream gz file for one of the datastreams is correct. You can do so by running the purge goal. Your issue is essentially #4165, so it should be fixed once your pipeline picks up an image with the just-release version 7.1.0 of DependencyCheck.

**Comment by morzan1001** [comment]
Hi, thanks for looking at this issue but I think you should reopen the issue because my problem is not solved by a suggested solution :( 

I am using the latest image in the pipeline (build about 11 hours ago) I have deleted all images, containers and volumes on the runner for testing purposes, so it is really clean and fetches everything again. Nothing is cached from Docker (yes this is not necessarily useful but the current setup). So the container throws the error described above even with the container in version 7.1.0. 

I also tried to perform a --purge in the container. To do this, before I run the dependency-check.sh with information about the project, I ran dependency-check.sh --purge once via pipelinecommand. But because the Docker container is brand new there is no database yet and the following output comes back: 

```
$ /usr/share/dependency-check/bin/dependency-check.sh --purge
[INFO] Unable to purge database; the database file does not exist: /usr/share/dependency-check/data/odc.mv.db
```

Do you have any other idea what I could try? I had read that the source error may have to do with a corrupted jar. By the way, I am using a virtualized Ubuntu with Docker 20.10.12 installed. 

Many greetings

**Comment by morzan1001** [comment]
Hey it's me again :D 

I have found the problem and solved it. You have updated versions of alpine or dotnet in your last commits. 

I have had a very similar problem with my Sonarqube instance that no longer ran after an Alpine update. The issue is here: https://github.com/SonarSource/docker-sonarqube/issues/544

Because everything is virtualized with a KVM-CPU there seems to be missing commands in the instruction set, which Java uses. I changed the CPU type to SandyBridge for my CPU (which matches the virtualization host) and tadaa it runs again.
